### ISSUE TITLE
[Enhancement] Curate the list of blocks that can be added to the Mini-Cart block

### ISSUE THREAD

**Issue by ralucaStan** [woothemes/woocommerce#46726]
## Problem
As a merchant, in the Editor I can add a lot of blocks to the Mini Cart. 

1. open the Editor
2. go to Patterns > Mini-Cart
3. in the Mini-Cart scroll to the bottom and select the + button to add a new inner block
4. notice one can add almost all blocks (even the Cart shortcode)
5. Try to add the Product search block; See that it's broken inside the Mini Cart

## Expected result
We need to allow only the blocks that make sense and work well for the Mini cart block. This is an important step in the checkout process. 

We can allow future Cross Sell and Upsell product blocks for example.

**Comment by pmcpinto** [comment]
> We need to allow only the blocks that make sense and work well for the Mini cart block. This is an important step in the checkout process.

I think we should start with a small subset of blocks and gradually introduce more. WDYT?

**Comment by nerrad** [comment]
@ralucaStan the [expanded use of `allowedBlocks` configuration property](https://make.wordpress.org/core/2024/03/09/miscellaneous-editor-changes-in-wordpress-6-5/#support-for-new-allowedblocks-field-in-block-json) for block.json (including being filterable) is something that should be used here.

**Comment by ralucaStan** [comment]
> I think we should start with a small subset of blocks and gradually introduce more. WDYT?

@pmcpinto  Agreed. We should define a list of allowed blocks first, before putting this ticket in the cycle. Should we start a GH discussion to track our decision there and collect some community suggestions?

> @ralucaStan the [expanded use of allowedBlocks configuration property](https://make.wordpress.org/core/2024/03/09/miscellaneous-editor-changes-in-wordpress-6-5/#support-for-new-allowedblocks-field-in-block-json) for block.json (including being filterable) is something that should be used here.

Thank you @nerrad , I'm aware of that API. For now I just wanted to start the discussion from a product point of view. Once we have the list of allowed blocks we can start discussing the implementation details.

**Comment by pierorocca** [comment]
@yjailin, would like to get your perspective on the idea of restricting blocks that can be inserted into the mini-cart.

The intent to curate a good experience is a good one. We can get to the desired outcome in a way that's more scalable and respectful of 3P devs than a curated allow list.

The scaling challenges with the proposal:

* We have to define what's "good" or "allowed" in a way that captures a broad range of use cases and across industries. I don't know how to do that in a way that doesn't exclude a legitimate block.
* It excludes a merchant from adding a custom block or at least puts the dependency on us to put it on the allowlist.
* It would require ongoing maintenance to maintain the curated list. That's not something any of us are going to take on.
* Customers aren't asking for this. On the contrary they're asking for more access to extend and to innovate more than we're allowing.

I can't rationalize prioritizing this but if we did, could having the block optionally declare what it's not compatible with (or that the dev doesn't want to support) be a feasible, self-serving, and scalable option?
